Source/JavaScriptCore/runtime/LiteralParser.h

```diff
@@ -151,25 +151,34 @@ class LiteralParser {
     
     JSValue tryLiteralParse()
     {
+        ASSERT(m_mode == StrictJSON);
+        ASSERT(reviverMode == JSONReviverMode::Disabled);
         m_lexer.next();
-        JSValue result;
         VM& vm = getVM(m_globalObject);
-        if (m_mode == StrictJSON && reviverMode == JSONReviverMode::Disabled)
-            result = parseRecursivelyEntry(vm);
-        else {
-            result = parse(vm, StartParseStatement, nullptr);
-            if (m_lexer.currentToken()->type == TokSemi)
-                m_lexer.next();
-        }
+        JSValue result = parseRecursivelyEntry(vm);
         if (m_lexer.currentToken()->type != TokEnd)
             return JSValue();
         return result;
     }
 
     JSValue tryLiteralParse(JSONRanges* sourceRanges)
     {
+        ASSERT(m_mode == StrictJSON);
+        ASSERT(reviverMode == JSONReviverMode::Enabled);
         m_lexer.next();
-        JSValue result = parse(getVM(m_globalObject), m_mode == StrictJSON ? StartParseExpression : StartParseStatement, sourceRanges);
+        JSValue result = parse(getVM(m_globalObject), StartParseExpression, sourceRanges);
```

Comment:
Can you explain the relationship between `m_mode == StrictJSON` and `StartParseExpression : StartParseStatement`?  Why does `StrictJSON` imply `Statement`?  What does `ParseStatement` mean in this context?  Thanks.

Reply:
`StartParseStatement` means it can accept `(` and `)`, which is not usable for JSON.